src/connector.rs

```diff
@@ -616,6 +626,37 @@ impl EthConnectorContract {
     pub fn set_paused_flags(&mut self, args: PauseEthConnectorCallArgs) {
         self.set_paused(args.paused_mask);
     }
+
+    /// Return metdata
+    pub fn get_metadata(&self) {
+        let icon = if let Some(ref icon) = self.metadata.icon {
+            format!(r#""{}""#, icon)
+        } else {
+            "null".to_string()
+        };
```

Comment:
Nit: Could avoid allocations here by using `Cow<str>`. Should also probably factor this out into a function like

```rust
fn string_or_null<T: Display>(data: Option<&T>) -> Cow<str> { ... }
```

Reply:
Looks like overoptimization.